webapp/platform/components/src/generic_modal/generic_modal.tsx

```diff
@@ -184,13 +206,24 @@ export class GenericModal extends React.PureComponent<Props, State> {
             </div>
         );
 
+        const locationClassMapping: Record<Required<Props>['modalLocation'], string> = {
+            top: 'GenericModal__location--top',
+            center: 'GenericModal__location--center',
+            bottom: 'GenericModal__location--bottom',
+        };
+
+        const modalLocationClass = locationClassMapping[this.props.modalLocation ?? 'center'];
+        const ariaLabeledBy = this.props.ariaLabeledBy || 'genericModalLabel';
```

Comment:
If we're always specifying the `aria-labelledby` now, I think that's going to make it so that the `aria-label` is only used when this element doesn't exist. Is that what you're intending here?

Reply:
Ok, so yes, the idea is that the current implementation will always set aria-labelledby (either to a custom value or default to 'genericModalLabel'), which means aria-label would be a fallback and only be used if the referenced element doesn't exist.

This was intentional because:

- We always want to have a fallback labeling mechanism
- We should prefer using aria-labelledby when possible as it's more dynamic (it reads the actual content of the referenced element)
- The 'genericModalLabel' element should always exist in our modal structure

I also added some comments to make the code clearer.